src/go/kreq-gen/kafka-request-generator.go

```diff
@@ -0,0 +1,239 @@
+	var tags kmsg.Tags
+	n := uint32(rand.Intn(10))
+	for i := uint32(0); i < n; i++ {
+		/// Make random bytes
+		token := make([]byte, rand.Intn(50))
+		rand.Read(token)
+		/// Choosing 100 as an arbitrary number so the tag IDs of unknown tags
+		/// never clash with those of known tags. Upon performing an audit the
+		/// largest tag id is < 10
+		tags.Set(100+i, token)
+	}
+	return tags
+}
+
+func makeRandomLegacyBatchV0() []kmsg.MessageV0 {
+	n := rand.Intn(10)
+	records := make([]kmsg.MessageV0, n)
+	for i := 0; i < n; i++ {
+		legacyRecord := kmsg.NewMessageV0()
+		randomFillType(&legacyRecord, int16(-1))
+		legacyRecord.Magic = int8(0)
+		legacyRecord.Attributes = int8(0)
+		serialized := legacyRecord.AppendTo(nil)
+		legacyRecord.CRC = int32(crc32.ChecksumIEEE(serialized[8+4+4:]))
+		legacyRecord.MessageSize = int32(len(serialized[8+4:]))
+		records[i] = legacyRecord
+	}
+	return records
+}
+
+func makeRandomLegacyBatchV1() []kmsg.MessageV1 {
+	n := rand.Intn(10)
+	records := make([]kmsg.MessageV1, n)
+	for i := 0; i < n; i++ {
+		legacyRecord := kmsg.NewMessageV1()
+		randomFillType(&legacyRecord, int16(-1))
+		legacyRecord.Magic = int8(1)
+		legacyRecord.Attributes = int8(0)
+		serialized := legacyRecord.AppendTo(nil)
+		legacyRecord.CRC = int32(crc32.ChecksumIEEE(serialized[8+4+4:]))
+		legacyRecord.MessageSize = int32(len(serialized[8+4:]))
+		records[i] = legacyRecord
+	}
+	return records
+}
+
+func makeRandomRecords() []kmsg.Record {
+	n := rand.Intn(10)
+	records := make([]kmsg.Record, n)
+	for i := 0; i < n; i++ {
+		record := kmsg.NewRecord()
+		randomFillType(&record, int16(-1))
+		recordSize := len(record.AppendTo(nil))
+		record.Length = int32(recordSize) - 4
+		records[i] = record
+	}
+	return records
+}
+
+func makeRandomRecordBatches() kmsg.RecordBatch {
+	recordBatch := kmsg.NewRecordBatch()
+	randomFillType(&recordBatch, int16(-1))
+	records := makeRandomRecords()
+	var bytearr []byte
+	for _, record := range records {
+		bytearr = append(bytearr, record.AppendTo(nil)...)
+	}
+	recordBatch.Records = bytearr
+	recordBatch.NumRecords = int32(len(records))
+	recordBatch.Magic = int8(2)       // new format
+	recordBatch.Attributes = int16(0) // uncompressed
+	rawBatch := recordBatch.AppendTo(nil)
+	recordBatch.Length = int32(len(rawBatch[8+4:]))
+	// skip first offset (int64) and length
+	recordBatch.CRC = int32(crc32.Checksum(rawBatch[8+4+4+1+4:], crc32c))
+	return recordBatch
+}
+
+func makeRandomKafkaData(version int16) []byte {
+	var bytearr []byte
+	if version >= 3 {
+		data := makeRandomRecordBatches()
+		bytearr = data.AppendTo(nil)
+	} else if version == 2 {
+		data := makeRandomLegacyBatchV1()
+		for _, record := range data {
+			bytearr = append(bytearr, record.AppendTo(nil)...)
+		}
+	} else {
+		data := makeRandomLegacyBatchV0()
+		for _, record := range data {
+			bytearr = append(bytearr, record.AppendTo(nil)...)
+		}
+	}
+	return bytearr
+}
+
+func randomFill(v reflect.Value, version int16) {
```

Comment:
what if you use this https://github.com/brianvoe/gofakeit

`gofakeit` allows you to define more interesting random strings, bytes, chars, names, addresses, etc you can even give it a spec and rangers, etc.